packages/shared/src/hooks/useViewSize.ts

```diff
@@ -0,0 +1,48 @@
+import { useMemo } from 'react';
+import { useMedia } from './useMedia';
+import {
+  desktop,
+  desktopL,
+  laptop,
+  laptopL,
+  laptopXL,
+  mobileL,
+  tablet,
+} from '../styles/media';
+
+export enum ViewSize {
+  MobileM = 'mobileM',
+  MobileL = 'mobileL',
+  Tablet = 'tablet',
+  Laptop = 'laptop',
+  LaptopL = 'laptopL',
+  LaptopXL = 'laptopXL',
+  Desktop = 'desktop',
+  DesktopL = 'desktopL',
+}
+
+const reversedEvaluatedSizes = [ViewSize.MobileM, ViewSize.MobileL];
```

Comment:
What is this for?

Reply:
So for those two we actually check if they are NOT one of those.
EG: MobileL checks if it's not Tablet.

This is a downside of us using `min-width` if we were to add `max-width` as types we could do it, but i was a bit hesitant as it also affect CSS usage.

Hope that explains it well?

You can see the final use in the return where we either return the check or the negated check.

Which is the same as this from the previous:
```
  const isMobile = !useMedia([tablet.replace('@media ', '')], [true], false);
  const isTablet = !useMedia([laptop.replace('@media ', '')], [true], false);
  const isLaptop = useMedia([laptop.replace('@media ', '')], [true], false);
```

Reply:
Hmm, would negation on the hook just solve it `!useViewSize(mobileM)`? Or more verbose:
```js
const isMobile = useViewSize(mobileM) === false
```

Feels wrong to have that kind of internal logic that is not obvious from hooks public interface.

Reply:
Wont work as the evaluation is against another query basically.
The only solution is i think to use max width then, but we already go against this pattern now

Reply:
I personally think below is clear:
```js
const isMobile = useViewSize(mobileM) === false
```

It's just comparing two boolean values in javascript? Let's ask others, I am fine with going either way.

Reply:
Let's wait for others indeed.
From my point of view it's a bit unclear you have to do this:

```js
// To evaluate if isMobile is true
const isMobile = useViewSize(mobileM) === false

// To evaluate if tablet is true
const isTablet = useViewSize(tablet)
```

I feel like for someone new you wouldn't know you need to manually negate for mobile sizes right?
This is why I wanted to solve it inside the hook so the return values are reflecting what you are requesting.

Reply:
> I feel like for someone new you wouldn't know you need to manually negate for mobile sizes right?

Same thoughts. Hence, I would much prefer if it would automatically be ready right from the instance call.

Reply:
I am on the same boat that we should not expose internal implementation on the public interface of the function. Let's keep it consistent?

Reply:
@DragosIorgulescu Not sure I get the right idea, you like the proposed idea or Ante's idea?

Reply:
@capJavert Are you ok with keeping it inside the hook as proposed?
See discussion above.

Reply:
🚀